pkg/pillar/cmd/zedmanager/zedmanager.go

```diff
@@ -783,53 +607,24 @@ func quantifyChanges(config types.AppInstanceConfig,
 	var purgeReason, restartReason string
 	log.Infof("quantifyChanges for %s %s",
 		config.Key(), config.DisplayName)
-	if len(status.StorageStatusList) != len(config.StorageConfigList) {
-		str := fmt.Sprintf("number of volumes changed from %d to %d",
-			len(status.StorageStatusList),
-			len(config.StorageConfigList))
+	if len(status.VolumeRefStatusList) != len(config.VolumeRefConfigList) {
+		str := fmt.Sprintf("number of volume ref changed from %d to %d",
+			len(status.VolumeRefStatusList),
+			len(config.VolumeRefConfigList))
 		log.Infof(str)
 		needPurge = true
 		purgeReason += str + "\n"
 	} else {
-		for _, sc := range config.StorageConfigList {
-			ss := lookupStorageStatus(&status, sc)
-			if ss == nil {
-				str := fmt.Sprintf("missing StorageStatus for (Name: %s, "+
-					"ImageSha256: %s, ImageID: %s, PurgeCounter: %d)",
-					sc.Name, sc.ImageSha256, sc.ImageID, sc.PurgeCounter)
+		for _, vrc := range config.VolumeRefConfigList {
+			vrs := lookupVolumeRefStatus(&status, vrc)
+			if vrs == nil {
+				str := fmt.Sprintf("Missing VolumeRefStatus for (VolumeID: %s, GenerationCounter: %d)",
+					vrc.VolumeID, vrc.GenerationCounter)
```

Comment:
When a new volume is created, and the app is told to use the new volume ref, presumably the issue to report to the controller is that the volumeRef changed  %s#%d. (The old error had the same issue.)